samples/matter/common/src/bridge/util/bridge_util.h

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#pragma once
+
+#include "bridged_device.h"
+
+#include <cstdint>
+#include <functional>
+#include <map>
+
+/*
+   DeviceFactory template container allows to instantiate a map which
+   binds supported BridgedDevice::DeviceType types with corresponding
+   creation method (e.g. constructor invocation). DeviceFactory can
+   only be constructed by passing a user-defined initialized list with
+   <BridgedDevice::DeviceType, ConcreteDeviceCreator> pairs.
+   Then, Create() method can be used to obtain an instance of demanded
+   device type with all passed arguments forwarded to the underlying
+   ConcreteDeviceCreator.
+*/
+template <typename T, typename... Args> class DeviceFactory {
+public:
+	using ConcreteDeviceCreator = std::function<T *(Args...)>;
+	using CreationMap = std::map<BridgedDevice::DeviceType, ConcreteDeviceCreator>;
```

Comment:
Can we avoid using costly structures for containers that will likely have just a couple of elements? Probably a static array of key-value pairs would be good enough and required less memory.

Also, the current design introduces a reverse dependency - Bridge Manager includes a file from the bridge application. If you want to have a generic bridge module, why not to introduce a factory interface such as:
```
template <class T, class... Args>
class GenericFactory
{
    virtual T* Create(Args&&... args) = 0;
}

using GenericDeviceFactory = GenericFactory<BridgedDevice, BridgedDevice::DeviceType, const char* /* label */>;
...
```
and then an application would create a concrete implementation of the Create method that could be a simple switch-case (without the need for the costly map) and pass that implementation to the bridge manager upon initialization. Otherwise, it may be difficult to use the common bridge module if one must provide specific header files etc. in the application.

Reply:
We discussed f2f that there's a task to rework this to avoid using STL so feel free to ignore this comment for now.

Reply:
For now I will move the shell module to the `applications/matter_bridge` as it's suppose to be a part of the application.